services/QuillLMS/app/models/feedback_history.rb

```diff
@@ -358,4 +362,45 @@ def self.serialize_detail_by_activity_session(feedback_session_uid)
     output.symbolize_keys
   end
   # rubocop:enable Metrics/CyclomaticComplexity
+
+  # rubocop:disable Metrics/CyclomaticComplexity
+  def self.session_data_for_csv(activity_id: nil, start_date: nil, end_date: nil, filter_type: nil, responses_for_scoring: false)
+    sub_query = select(
```

Comment:
this method is worth breaking up a bit. You can make this sub_query it's own method, e.g.

```ruby
session_uids = session_data_uids(activity_id: activity_id, filter_type: filter_type, responses_for_scoring: responses_for_scoring)
...
query = query.where(feedback_session_uid: session_uids) if responses_for_scoring || filter_type


...
def self.session_data_uids(activity_id: nil, filter_type: nil, responses_for_scoring: false)
  ...

Reply:
agreed, I'll break this out into two methods